src/internal/endpoint-runtime.ts

```diff
@@ -0,0 +1,143 @@
+import type { JsonValue } from 'type-fest'
+import uuid from 'tiny-uid'
+import { serializeError } from 'serialize-error'
+import type { DataTypeKey, Destination, GetDataType, GetReturnType, IBridgeMessage, IInternalMessage, OnMessageCallback, RuntimeContext } from '../types'
+import { parseEndpoint } from './endpoint'
+
+export interface EndpointRuntime {
+  sendMessage: <ReturnType extends JsonValue, K extends DataTypeKey = DataTypeKey>(
```

Comment:
When extending `ProtocolMap` this `ReturnType` generic doesn't make sense because it's merely a fallback value used when `ProtocolMap` is not extended. I would recommend removing it or flipping the order of those type params. 

Having this as a generic here doesn't really buy us any safety whatsoever, it's just a "sugar" for:
```ts
sendMessage('foo', { test: 1 }).then((a: MyReturnValue) => {})
```
```